src/core/control/xojfile/LoadHandler.h

```diff
@@ -58,119 +47,102 @@ class LoadHandler {
 public:
     std::unique_ptr<Document> loadDocument(fs::path const& filepath);
 
-    std::string getLastError();
+    bool hasErrorMessages() const;
+    std::string getErrorMessages() const;
     bool isAttachedPdfMissing() const;
-    std::string getMissingPdfFilename() const;
+    fs::path getMissingPdfFilename() const;
 
     /** @return The version of the loaded file */
     int getFileVersion() const;
 
-private:
-    void parseStart();
-    void parseContents();
-    void parsePage();
-    void parseLayer();
-    void parseAudio();
-
-    void parseStroke();
-    void parseText();
-    void parseImage();
-    void parseTexImage();
+public:
+    // interface for XmlParser
+    void addXournal(const std::string& creator, int fileversion);
+    void addMrWriter(const std::string& creator);
+    void finalizeDocument();
+    void addPage(double width, double height);
+    void finalizePage();
+    void addAudioAttachment(const fs::path& filename);
+    void addBackground(const std::optional<std::string>& name);
+    void setBgSolid(const PageType& bg, const Color& color);
+    void setBgPixmap(bool attach, const fs::path& filename);
+    void setBgPixmapCloned(size_t pageNr);
+    void setBgPdf(size_t pageno);
+    void loadBgPdf(bool attach, const fs::path& filename);
+    void addLayer(const std::optional<std::string>& name);
```

Comment:
Yes, an empty string would be a valid layer name, however, as long as `setName()` is not called on the layer, it will be displayed as "Layer <#id>". Why would you suggest defaulting to nullopt? If no name attribute is given, that's what will be passed anyways.